frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,307 @@
+  constructor(props) {
+    super(props);
+
+    const {items} = this.props;
+
+    const filters = getFilters(items, filterCategories);
+
+    const filterCounts = initFilterCounts(items, filters);
+
+    this.state = {
+      filteredItems: items,
+      filters,
+      filterCounts,
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { filters, filterCounts } = this.state;
+    const { items } = this.props;
+
+    // Update filters to match new items
+    if (items !== prevProps.items) {
+      const newFilters = getFilters(items, filterCategories);
+      this.setState({filters: newFilters});
+    }
+
+    // Update tile display according to active filters
+    if (filters !== prevState.filters) {
+      const filteredItems = this.filterItems();
+      this.setState({filteredItems});
+    }
+
+    // filter counts are updated when filter by name changed
+    if (prevState.filters.byName !== filters.byName) {
+      this.setState({filterCounts: this.getFilterCounts(filters, filterCounts)});
+    }
+  }
+
+  getFilterCounts(filters, filterCounts) {
+    const {items} = this.props;
+    const newFilterCounts = {...filterCounts};
+
+    // Counts are based off of text filtered items
+    const filteredItems = this.filterByName(items);
+
+    for (const key in filters){
+      if (key === 'byName') {
+        continue;
+      }
+      let itemKey = key;
+      if (itemKey.startsWith('by')){
+        itemKey = itemKey.replace(/by/, '');
+      }
+
+      const count = _.countBy(filteredItems, itemKey);
+
+      for (const elem in newFilterCounts[key]) {
+        if (newFilterCounts[key].hasOwnProperty(elem)) {
+          newFilterCounts[key][elem] = count[elem] || 0;
+        }
+      }
+    }
+
+    return newFilterCounts;
+  }
+
+  filterByName(items) {
+    const { filters } = this.state;
+    const { byName } = filters;
+    if (byName.active) {
+      const filterString = byName.value.toLowerCase();
+      return _.filter(( items ), item => item.tileName.toLowerCase().includes(filterString));
+    }
+    return items;
+  }
+
+  filterItems() {
+    const { filters } = this.state;
+    const { items } = this.props;
+
+    let filteredItems = items;
+
+    // Filters as follows:
+    //   Active filters in the same category result in the union of items shown
+    //   Active filters in separate categories result in the intersection
+    for (const key in filters){
+      if (key === 'byName') {
+        continue;
+      }
+      let itemKey = key;
+      if (itemKey.startsWith('by')){
+        itemKey = itemKey.replace(/by/, '');
+      }
+
+      let categoryItems = [];
+      let categoryActive = false;
+      const category = filters[key];
+      for (const elem in category) {
+        if (category[elem].active) {
+          categoryActive = true;
+          categoryItems = _.union(categoryItems, _.filter(filteredItems, (item) => item[itemKey].includes(category[elem].value)));
```

Comment:
Checkboxes filter by comparing each attribute with `includes` rather than equality, which is undesirable for some categories.